llvm/lib/Target/SBF/SBFInstrInfo.td

```diff
@@ -942,8 +970,32 @@ let isCodeGenOnly = 1 in {
   def MOV_32_64 : ALU_RR<SBF_ALU, SBF_MOV,
                          (outs GPR:$dst), (ins GPR32:$src),
                          "mov32 $dst, $src", []>;
+  let Predicates = [SBFv2] in {
+    def MOV_32_64_imm : ALU_RI<SBF_ALU, SBF_MOV,
+                               (outs GPR:$dst), (ins i32imm:$imm),
+                               "mov32 $dst, $imm", []>;
+    def MOV_32_64_addr : ALU_RI<SBF_ALU, SBF_MOV,
+                           (outs GPR:$dst), (ins u64imm:$imm),
+                           "mov32 $dst, $imm", []>;
+  }
 }
 
+// In SBFv2, a CopyToReg of a 64-bit value is split in two instructions:
+// mov32 r1, 0x55667788
+// hor r1, 0x11223344
+// These instructions copy the value 0x1122334455667788 to a register.
+def : Pat<(i64 imm:$imm),
+            (HOR (MOV_32_64_imm (i32 (Lower64 $imm))),
+                       (i32 (Upper64 $imm)))>, Requires<[SBFv2]>;
```

Comment:
Note that since we now generate the pair rather than `lddw`, there are opportunities for minor optimizations here. For example, if the upper 32-bits is clear, we only need to emit one instruction (and similar patterns). 

It isn't important for this patch, but a follow-up patch might add those tweaks and corresponding unit tests. Given the absurdly long instruction word size of BPF/SBF (and the code size restrictions of Solana contracts/programs), eliminating redundant instructions like this is a good idea.

Reply:
I'll do this in a follow-up PR.